### PR TITLE
CIC Vendor Expansion (Part 2)

### PR DESCRIPTION
**Pull request by SpypigDev**
# About The Pull Request
Follow up PR to [#4707](https://github.com/cmss13-devs/cmss13/pull/4707). Aims to expand XO vendors to bring them up to date with modern combat meta, as well as clean up my SO vendor update in the process.
- Added gear specializations to the XO weapons vendor, offering the essentials set of either a Medic or Engineer
- Adds a Magharn to rail attachments section in the XO weapons vendor
- Expanded belt/pouch options in the XO weapons vendor
- Added the 'other gear' section in the XO vendor, with various combat supplies

Misc.
- Fixed XO/SO vendor code so that selecting the autoinjector pouch gives you a full one, rather than empty
- ~~Increased XO policing skillset from 1, to 2, so they can actually use the MP belt they are offered~~ (Already done in #6781 )
- Unified the SO armory vendor code into the regular SO vendor file, rather than having two seperate

# Explain why it's good for the game

The XO gear vendors have sat untouched in the code for quite some time, and have been left to fall behind the quickly changing meta of CM. With a more updated selection of modern gear, XOs will no longer be significantly out-geared by their Staff Officers when it comes to organizing their combat kits, especially during hijack. 

XOs will be brought up to par with the gear diversity that SOs have enjoyed for some months now.

# Testing Photographs and Procedure

Select photos of the updated XO vendor menus, where changes have been made

<details>

Screenshot 2024-05-17 160207

Screenshot 2024-05-17 160216

Screenshot 2024-05-17 160228

Screenshot 2024-05-17 160236

</details>

# Changelog
:cl:
refactor: SO armory vendor code refactored into the main SO gear vendor code
balance: XO weapon and clothing vendors fitted with a more modernized equipment set
balance: XO weapon vendor now offers Medic or Engineer essentials sets
fix: Vending an autoinjector pouch as SO or XO now produces a full pouch, rather than empty
balance: SO's mod88 removed from cryo spawn and moved to their vendor instead
/:cl:

### CODE REVIEW

File: maps/map_files/USS_Almayer/USS_Almayer.dmm
Comment (Nanu308) [inline maps/map_files/USS_Almayer/USS_Almayer.dmm:1]: mapping itself is fine

**Review by Nanu308 — COMMENTED** [review]
mapping itself is fine

**Comment by Drulikar** [comment]
Stale exempt until it can be balance/code reviewed. Sorry just a lot going on right now.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Review by Drulikar — CHANGES_REQUESTED** [review]
Feel free to undraft when you're happy with the vendor contents. If both are going to have essential kits, likely their vendors should be very similar just minus anything XO might get in their other vendor.

**Review by Drulikar — CHANGES_REQUESTED** [review]
Mentioned in DMs: 
- Consider adding `/obj/item/storage/belt/gun/utility/full` to XO since SO is offered it.
- Consider adding `/obj/item/attachable/bayonet` to XO since SO is offered it.
- Consider removing `/obj/item/storage/belt/gun/m4a3/mod88` from SO cryo spawn to instead offer it with with their other pistol belt options in their vendor.

**Comment by deathrobotpunch** [comment]
one thing I would like to see is a flash added to the SO vendor like every other head of staff.

**Review by Drulikar — APPROVED** [review]

File: code/game/machinery/vending/vendor_types/crew/senior_officers.dm [hunk 1]
```diff
@@ -393,53 +417,60 @@ GLOBAL_LIST_INIT(cm_vending_clothing_xo, list(
...
+		list("Falling Falcons Shoulder Patch", 1, /obj/item/clothing/accessory/patch/falcon, null, VENDOR_ITEM_REGULAR),
+		list("USCM Shoulder Patch", 1, /obj/item/clothing/accessory/patch, null, VENDOR_ITEM_REGULAR),
+
+
+		list("BELT (CHOOSE 1)", 0, null, null, null),
+		list("G8-A General Utility Pouch", 0, /obj/item/storage/backpack/general_belt, MARINE_CAN_BUY_BELT, VENDOR_ITEM_REGULAR),
+		list("M276 Ammo Load Rig", 0, /obj/item/storage/belt/marine, MARINE_CAN_BUY_BELT, VENDOR_ITEM_RECOMMENDED),
+		list("M276 Shotgun Shell Loading Rig", 0, /obj/item/storage/belt/shotgun, MARINE_CAN_BUY_BELT, VENDOR_ITEM_RECOMMENDED),
+		list("M276 Toolbelt Rig (Full)", 0, /obj/item/storage/belt/utility/full, MARINE_CAN_BUY_BELT, VENDOR_ITEM_RECOMMENDED),
+		list("M276 Holster Toolrig", 0, /obj/item/storage/belt/gun/utility/full, MARINE_CAN_BUY_BELT, VENDOR_ITEM_REGULAR),
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/crew/senior_officers.dm:442]:
```suggestion
		list("M276 Holster Toolrig (Full)", 0, /obj/item/storage/belt/gun/utility/full, MARINE_CAN_BUY_BELT, VENDOR_ITEM_REGULAR),
```

File: code/game/machinery/vending/vendor_types/crew/staff_officer.dm [hunk 2]
```diff
@@ -46,8 +58,85 @@ GLOBAL_LIST_INIT(cm_vending_clothing_staff_officer, list(
...
+
+		list("BELT (CHOOSE 1)", 0, null, null, null),
+		list("G8-A General Utility Pouch", 0, /obj/item/storage/backpack/general_belt, MARINE_CAN_BUY_BELT, VENDOR_ITEM_REGULAR),
+		list("M276 Ammo Load Rig", 0, /obj/item/storage/belt/marine, MARINE_CAN_BUY_BELT, VENDOR_ITEM_RECOMMENDED),
+		list("M276 Shotgun Shell Loading Rig", 0, /obj/item/storage/belt/shotgun, MARINE_CAN_BUY_BELT, VENDOR_ITEM_RECOMMENDED),
+		list("M276 Toolbelt Rig (Full)", 0, /obj/item/storage/belt/utility/full, MARINE_CAN_BUY_BELT, VENDOR_ITEM_RECOMMENDED),
+		list("M276 Lifesaver Bag (Full)", 0, /obj/item/storage/belt/medical/lifesaver/full, MARINE_CAN_BUY_BELT, VENDOR_ITEM_RECOMMENDED),
+		list("M276 Medical Storage Rig (Full)", 0, /obj/item/storage/belt/medical/full, MARINE_CAN_BUY_BELT, VENDOR_ITEM_REGULAR),
+		list("M276 M39 Holster Rig", 0, /obj/item/storage/belt/gun/m39, MARINE_CAN_BUY_BELT, VENDOR_ITEM_REGULAR),
+		list("M276 Holster Toolrig", 0, /obj/item/storage/belt/gun/utility/full, MARINE_CAN_BUY_BELT, VENDOR_ITEM_REGULAR),
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/crew/staff_officer.dm:101]:
```suggestion
		list("M276 Holster Toolrig (Full)", 0, /obj/item/storage/belt/gun/utility/full, MARINE_CAN_BUY_BELT, VENDOR_ITEM_REGULAR),
```

**Review by Drulikar — APPROVED** [review]

**Pull request merged**